Modules/Chatroom/classes/gui/class.ilChatroomSettingsGUI.php

```diff
@@ -29,50 +29,57 @@ class ilChatroomSettingsGUI extends ilChatroomGUIHandler
+        );
+
+        $result = (new \ILIAS\Data\Factory())->error($this->ilLng->txt('form_input_not_valid'));
+        if ($this->http->request()->getMethod() === 'POST') {
+            $settingsForm = $settingsForm->withRequest($this->http->request());
+            $result = $settingsForm->getInputGroup()->getContent();
+        }
+
+        if (!$result->isOK()) {
+            $this->mainTpl->setOnScreenMessage('failure', $result->error());
             $this->general($settingsForm);
-        } else {
-            $this->gui->getObject()->setTitle($settingsForm->getInput('title'));
-            $this->gui->getObject()->setDescription($settingsForm->getInput('desc'));
-
-            /** @var ilDateDurationInputGUI $period */
-            $period = $settingsForm->getItemByPostVar('access_period');
-            if ($period->getStart() && $period->getEnd()) {
-                $this->gui->getObject()->setAccessType(ilObjectActivation::TIMINGS_ACTIVATION);
-                $this->gui->getObject()->setAccessBegin($period->getStart()->get(IL_CAL_UNIX));
-                $this->gui->getObject()->setAccessEnd($period->getEnd()->get(IL_CAL_UNIX));
-                $this->gui->getObject()->setAccessVisibility((int) $settingsForm->getInput('access_visibility'));
-            } else {
-                $this->gui->getObject()->setAccessType(ilObjectActivation::TIMINGS_DEACTIVATED);
-            }
+            return;
+        }
 
-            $this->gui->getObject()->update();
-            $this->obj_service->commonSettings()->legacyForm($settingsForm, $this->gui->getObject())->saveTileImage();
+        $values = $result->value();
 
-            $room = ilChatroom::byObjectId($this->gui->getObject()->getId());
-            $requestSettings = $room->getSettings();
-            if (!$room) {
-                $room = new ilChatroom();
-                $requestSettings['object_id'] = $this->gui->getObject()->getId();
-            }
+        $this->gui->getObject()->getObjectProperties()->storePropertyTitleAndDescription(
+            $values[ilChatroomFormFactory::PROP_TITLE_AND_DESC]
+        );
+        $this->gui->getObject()->getObjectProperties()->storePropertyIsOnline(
+            $values[ilChatroomFormFactory::PROP_ONLINE_STATUS]
+        );
+        $this->gui->getObject()->getObjectProperties()->storePropertyTileImage(
+            $values[ilChatroomFormFactory::PROP_TILE_IMAGE]
+        );
 
-            foreach ($requestSettings as $setting => &$value) {
-                if ($settingsForm->getItemByPostVar($setting) !== null) {
-                    $value = $settingsForm->getInput($setting);
-                }
+        $room = ilChatroom::byObjectId($this->gui->getObject()->getId());
+        $mutated_settings = $room->getSettings();
+        if (!$room) {
+            $room = new ilChatroom();
+            $mutated_settings['object_id'] = $this->gui->getObject()->getId();
+        }
+
+        foreach ($mutated_settings as $setting => &$value) {
```

Comment:
Can be simplified with:
```suggestion
        $mutated_settings = array_merge($mutated_settings, array_intersect($mutated_settings, $values));
```